src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs

```diff
@@ -1289,6 +1292,10 @@ protected override Expression VisitLambda<T>(Expression<T> node)
                 }
             }
             Visit(body);
+            
+            if (body.NodeType == ExpressionType.MemberAccess && body.Type.GetInterfaces().Any(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IDictionary<,>)))
+                Out(".ToDictionary(e1 => e1.Key, e1 => e1.Value)");
```

Comment:
Same as above

Reply:
Changed

Reply:
Changed to `e3` so it doesn't potentially shadow variable named `x` that user could want to use